DataFormats/HcalRecHit/src/HBHERecHit.cc

```diff
@@ -62,3 +62,22 @@ void HBHERecHit::getMergedIds(std::vector<HcalDetId>* ids) const
         }
     }
 }
+
+
+HcalDetId HBHERecHit::idFront() const {
+  if (isMerged()) {
+    HcalDetId myId(id());
+    if (auxPhase1_ & (1U << HBHERecHitAuxSetter::OFF_COMBINED)) {
+      const unsigned nMerged = CaloRecHitAuxSetter::getField(auxPhase1_,
+							     HBHERecHitAuxSetter::MASK_NSAMPLES,
+							     HBHERecHitAuxSetter::OFF_NSAMPLES);
+      if (nMerged > 0) {
+	const unsigned depth = CaloRecHitAuxSetter::getField(auxHBHE_, 0xf, 0);
+	myId = HcalDetId(myId.subdet(), myId.ieta(), myId.iphi(), depth);
```

Comment:
@bsunanda
Sorry, the order in which the depths are packed in the merged rechit is undefined. If you need the smallest one, it should be found explicitly. Alternatively, if you think that it would be better to have them in the increasing order, I can change the code which fills them